compiler/rustc_middle/src/mir/mono.rs

```diff
@@ -468,6 +469,20 @@ impl<'tcx> CodegenUnit<'tcx> {
         hash.as_u128().to_base_fixed_len(CASE_INSENSITIVE)
     }
 
+    pub fn shorten_name(human_readable_name: &str) -> Cow<'_, str> {
+        // Set a limit a somewhat below the common platform limits for file names.
+        const MAX_CGU_NAME_LENGTH: usize = 200;
```

Comment:
Is this something we can query from the OS?

Reply:
This needs to be less than the actual filename limit, since the extension and other bits are added later.

Reply:
Browsing the limits it looks like 255 is by far the most common amongst filesystems https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits